nevergrad/functions/ttp/core.py

```diff
@@ -0,0 +1,224 @@
+                line = line.strip()
+                self.numberOfItems = int(line)
+
+            if line.startswith("CAPACITY OF KNAPSACK"):
+                line = line[line.index(":") + 1 :]
+                line = line.strip()
+                self.capacityOfKnapsack = int(line)
+
+            if line.startswith("MIN SPEED"):
+                line = line[line.index(":") + 1 :]
+                line = line.strip()
+                self.minSpeed = float(line)
+
+            if line.startswith("MAX SPEED"):
+                line = line[line.index(":") + 1 :]
+                line = line.strip()
+                self.maxSpeed = float(line)
+
+            if line.startswith("RENTING RATIO"):
+                line = line[line.index(":") + 1 :]
+                line = line.strip()
+                self.rentingRatio = float(line)
+
+            if line.startswith("EDGE_WEIGHT_TYPE"):
+                line = line[line.index(":") + 1 :]
+                line = line.strip()
+                self.edgeWeightType = line
+
+            if line.startswith("NODE_COORD_SECTION"):
+                self.nodes = []
+
+                for i in range(0, self.numberOfNodes):
+                    count += 1 + i - i
+                    line = lines[count]
+                    splittedLine = line.split()
+                    col = []
+
+                    for j in range(0, len(splittedLine)):
+                        temp = float(splittedLine[j])
+                        if j == 0:
+                            temp = temp - 1
+                        col.append(temp)
+                    self.nodes.append(col)
+
+            if line.startswith("ITEMS SECTION"):
+                self.items = []
+                for i in range(0, self.numberOfItems):
+                    count += 1
+                    line = lines[count]
+                    splittedLine = line.split()
+                    col = []
+
+                    for j in range(0, len(splittedLine)):
+                        temp = int(splittedLine[j])
+                        # adjust city number by 1
+                        if j == 0:
+                            temp = temp - 1
+                        if j == 3:
+                            temp = temp - 1
+                        col.append(temp)
+                    self.items.append(col)
+
+            count += 1
+        reader.close()
+
+        init1 = np.zeros(self.numberOfNodes - 1)
+        p1 = p.Array(init=init1)
+        p1.set_bounds(0, 4)
+        p1.set_name("tspTour")
+
+        init2 = 0 * np.ones(self.numberOfItems)
+        p2 = p.Array(init=init2)
+        p2.set_integer_casting()
+        p2.set_bounds(0, 1)
+        p2.set_name("packingPlan")
+
+        instru = ng.p.Instrumentation(tspTour=p1, packingPlan=p2).set_name("")
+
+        super().__init__(self.evaluate, instru)
+
+    def evaluate(self, tspTour: np.ndarray, packingPlan: np.ndarray) -> float:
+
+        np.reshape(tspTour, (1, -1))

```

Comment:
instead of reshaping, you should make the parametrization have the right shape from the start, probably with
`init1 = np.zeros(1, self.numberOfNodes - 1)`

Reply:
Couldn't get get this or using choice for the packing plan to immediately work